### ISSUE TITLE
Support for Java 21 for azurerm_linux_web_app

### ISSUE THREAD

**Issue by christian-anker-larsen** [hashicorp/terraform-provider-azurerm#25490]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

As evidenced from the screenshot below taken in Azure Portal on 2nd April 2024, Azure Web Apps now support Java 21. For this reason, it would be great if Java 21 is supported in the `azurerm_linux_web_app` resource as well. Would it be possible to implement this any time in the foreseeable future? 🙏🏻 

<img width="518" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/61601217/c49dc6e1-0f1f-4d76-940c-82d2163f3e97">

Thank you.

### New or Affected Resource(s)/Data Source(s)

azurerm_linux_web_app

### Potential Terraform Configuration

```hcl
application_stack {
      java_version = 21
    }
```

**Comment by christian-anker-larsen** [comment]
I did some further investigation in Azure Portal, and it seems that even though `Java 21` is selectable in my web app's configuration, I am unable to save the changes. I then attempted to create a new Linux-based resource manually via Azure Portal with `Java 21`. As per the screenshot below, selecting this version disables the `Linux` option in the `Operating System` field. This seems to indicate that `Java 21` is not yet supported on Linux based Web Apps.

In other words, this issue is not an actual issue (yet), so I am closing it for now.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/61601217/444fe273-7eba-4a31-a013-7d891adc9199)

**Comment by ppa-fyayc** [comment]
The announcement is here: https://techcommunity.microsoft.com/t5/apps-on-azure-blog/announcing-java-21-and-tomcat-10-1-on-azure-app-service/ba-p/4105714

**Comment by ppa-fyayc** [comment]
Looks like it's tracked here https://github.com/hashicorp/terraform-provider-azurerm/issues/24754

**Comment by ppa-fyayc** [comment]
@christian-anker-larsen are you going to reopen this issue?

**Comment by christian-anker-larsen** [comment]
@ppa-fyayc I made another attempt in Azure Portal to create a Web App in Linux OS, but I am still seeing this, when selecting `Java 21`:

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/61601217/e4dec228-1f45-491f-95f1-0815838d4de0)

To me, it looks like that the announcement implicitly only applies for Windows OS. Not sure if it makes sense to reopen this issue until Linux is supported as well.

**Comment by ppa-fyayc** [comment]
Oh, I see what you mean :( That's a shame! Let's wait then.

**Comment by ppa-fyayc** [comment]
I was able to upgrade my linux web app to Java 21 using

resource "azapi_update_resource" "backend-webapp-java-21" {
  type        = "Microsoft.Web/sites@2023-01-01"
  resource_id = azurerm_linux_web_app.backend.id
  body        = jsonencode({
    properties = {
      siteConfig = {
        linuxFxVersion = "JAVA|21-java21"
      }
    }
  })
}

**Comment by iAMSagar44** [comment]
@christian-anker-larsen - I had encountered the same issue, but I can now see that Java 21 is supported for Azure App Service on Linux. 

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/29267138/a71e76d7-ad9e-4727-af20-7d4e2dd26b73)

@ppa-fyayc - I am unable to upgrade to Java 21 with the below terraform script -

```
site_config {
    application_stack {
      java_server         = "JAVA"
      java_version        = "21"
      java_server_version = "21"
    }
  }

```
It fails Terraform validation.

@christian-anker-larsen - I think this issue should be reopened. Thoughts?

**Comment by christian-anker-larsen** [comment]
@iAMSagar44 I just checked Azure Portal, and it's now possible to create Web Apps on Linux w/ Java 21. Also tried modifying an existing web app, and there it was also possible to change to Java 21, so, yes, I am reopening this issue.

**Comment by onclebendusud** [comment]
Good morning,
is there a plan(ning) for adding java21 in azurerm_linux_web_app ?
It's just to have an expected date for the availability.

**Comment by vidwah-nte** [comment]
Just to build on the work-around provided by @ppa-fyayc:
You might want to add the `replace_triggered_by` lifecycle to `azapi_update_resource`. I'm experiencing that if any fields in the `azurerm_linux_web_app` is modified (for instance, `site_config`), then the application stack is set to `JAVA|21-` and not `JAVA|21-java21`. `JAVA|21-` is an invalid value, and will most likely cause your app to stop working.
Here's a complete example for reference:
```
resource "azapi_update_resource" "backend-webapp-java-21" {
  type        = "Microsoft.Web/sites@2023-01-01"
  resource_id = azurerm_linux_web_app.backend.id
  body        = jsonencode({
    properties = {
      siteConfig = {
        linuxFxVersion = "JAVA|21-java21"
      }
    }
  })
  lifecycle {
    replace_triggered_by = [
      azurerm_linux_web_app.backend
    ]
  }
}
```
This will force `azapi_update_resource` to be recreated, setting the application stack back to `JAVA|21-java21`.

**Comment by sudeepvarma2017** [comment]
Is there any ETA for this issue ?

**Comment by MarcelTemp** [comment]
What is the status here?

**Comment by computerlove** [comment]
The status is that I have a [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/26304) that has a failing test, that I have no idea how to resolve.
